umpost/conversion_driver_esm1p5.py

```diff
@@ -110,14 +122,73 @@ def find_matching_fields_files(dir_contents, fields_file_name_pattern):
     return fields_file_paths
 
 
-def convert_fields_file_list(fields_file_paths, nc_write_dir):
+def get_nc_filename(fields_file_name, date=None):
+    """
+    Format a netCDF output filename based on the input fields file name and
+    its date. Assumes fields_file_name follows ESM1.5's naming convention
+    '{5 char run_id}.pa{unit}{date encoding}`.
+
+    Parameters
+    ----------
+    fields_file_name: name of fields file to be converted.
+    date: tuple of form (year, month, day) associated with fields file data,
+          or None. If None, ".nc" will concatenated to the original fields
+          file name.
+
+    Returns
+    -------
+    name: formated netCDF filename for writing output.
+    """
+    if date is None:
+        return f"{fields_file_name}.nc"
+
+    stem = fields_file_name[0:FF_UNIT_INDEX + 1]
+    unit = fields_file_name[FF_UNIT_INDEX]
+
+    try:
+        suffix = f"_{FF_UNIT_SUFFIX[unit]}"
+    except KeyError:
+        warnings.warn(
+            f"Unit code '{unit}' from filename f{fields_file_name} "
+            "not recognized. Frequency information will not be added "
+            "to the netCDF filename."
+        )
+        suffix = ""
+
+    year, month, _ = date
+    return f"{stem}-{year:04d}{month:02d}{suffix}.nc"
+
+
+def get_ff_date(fields_file_path):
     """
-    Convert group of fields files to NetCDF, writing output in nc_write_dir.
+    Get the date from a fields file. To be used for naming output files.
 
     Parameters
     ----------
-    fields_file_paths : list of paths to fields files for conversion.
-    nc_write_dir : directory to save NetCDF files into.
+    fields_file_path : path to single fields file.
+
+    Returns
+    -------
+    date_tuple : tuple of integers (yyyy,mm,dd) containing the fields
+                 file's date.
+    """
+    fields_file_header = mule.FixedLengthHeader.from_file(
+                                            str(fields_file_path))
+    year = fields_file_header.t2_year       # pylint: disable=no-member
+    month = fields_file_header.t2_month     # pylint: disable=no-member
+    day = fields_file_header.t2_day         # pylint: disable=no-member
+
+    return (year, month, day)
```

Comment:
One idea, we can simplify the code to deduplicate the `noqa` guards:
```suggestion
    header = mule.FixedLengthHeader.from_file(str(fields_file_path))
    return header.t2_year, header.t2_month, header.t2_day  # pylint: disable=no-member
```
This provides the benefit of constraining the guard to 1 line of code.

Reply:
Added this in 797a824